src/lib/protocol_state_validator/protocol_state_validator.ml

```diff
@@ -0,0 +1,60 @@
+open Core_kernel
+open Async_kernel
+open Protocols.Coda_pow
+open Protocols.Coda_transition_frontier
+open Coda_base
+
+module type Inputs_intf = sig
+  include Transition_frontier.Inputs_intf
+
+  module Time : Time_intf
+
+  module State_proof :
+    Proof_intf
+    with type input := Consensus.Mechanism.Protocol_state.value
+     and type t := Proof.t
+end
+
+let transition_error msg = Or_error.errorf !"transition rejected :%s" msg
+
+module Make (Inputs : Inputs_intf) :
+  Protocol_state_validator_intf
+  with type time := Inputs.Time.t
+   and type state_hash := State_hash.t
+   and type external_transition := Inputs.External_transition.t
+   and type external_transition_verified :=
+              Inputs.External_transition.Verified.t = struct
+  include Inputs
+
+  let to_unix_timestamp recieved_time =
+    recieved_time |> Time.to_span_since_epoch |> Time.Span.to_ms
+    |> Unix_timestamp.of_int64
+
+  let to_valid_protocol_state_transition unverified_transition =
```

Comment:
can you inline this function? I want to make sure the:

```
`I_swear_this_is_safe_see_my_comment
```

bit is always locally near the part of the code doing the verification.